### ISSUE TITLE
Wdeprecated-declarations from GCC v10.1.0

### ISSUE THREAD

**Issue by dpellegr** [nlohmann/json#2109]
Including `json.hpp v3.7.3` and compiling with `g++ v10.1.0` using `-std=c++20`, triggers the following warning:

```
json.hpp:12595:28: warning: ‘template<class _Tp> struct std::is_pod’ is deprecated: 
use is_standard_layout && is_trivial instead [-Wdeprecated-declarations]
12595 |         static_assert(std::is_pod<CharType>::value, "CharType must be POD");
```

**Comment by nlohmann** [comment]
Duplicate of #1913, and fixed in develop branch and next release 3.8.0.

**Comment by dpellegr** [comment]
Sorry, quickly I browsed the open issues, then I tried the wrong keywords for the closed ones and #1913 did not came up. Good that it has been done already.

**Comment by nlohmann** [comment]
No worries. It's time that 3.8.0 is released...
